components/docker-auth/pkg/auth/authn_handler.go

```diff
@@ -42,160 +41,164 @@ const (
+		}
+		if accessTokenValidity {
+			logger.Debugf("[%s] User successfully authenticated. Returning success status code", execId)
+			return true, nil
 		} else {
-			log.Printf("[%s] User failed to authenticate. Returning error status code\n", execId)
-			return extension.ErrorExitCode
+			logger.Debugf("[%s] User failed to authenticate. Returning error status code", execId)
+			return false, nil
 		}
 	}
 }
 
-func validateJWT(token string, username string, execId string) bool {
-	claim := getJWTClaims(token, execId)
-	iss := getClaimValue(claim, issuerClaim, execId)
-	sub := getClaimValue(claim, subjectClaim, execId)
+func validateJWT(token string, username string, logger *zap.SugaredLogger, execId string) (bool, error) {
+	claim := getJWTClaims(token, logger, execId)
+	iss := getClaimValue(claim, issuerClaim, logger, execId)
+	sub := getClaimValue(claim, subjectClaim, logger, execId)
 
-	log.Printf("[%s] Token issuer : %s\n", execId, iss)
-	log.Printf("[%s] Subject : %s\n", execId, sub)
+	logger.Debugf("[%s] Token issuer : %s", execId, iss)
+	logger.Debugf("[%s] Subject : %s", execId, sub)
 
 	if sub != username {
-		log.Printf("[%s] Username(%s) does not match with subject(%s) in JWT\n", execId, username, sub)
-		return false
+		logger.Debugf("[%s] Username(%s) does not match with subject(%s) in JWT", execId, username, sub)
+		return false, nil
 	}
 
-	certificateInUse, err := readCert(idpCertEnvVar, execId)
+	certificateInUse, err := readCert(idpCertEnvVar, logger, execId)
 	if err != nil {
-		log.Printf("[%s] Unable to load idp cert file : %s\n", execId, err)
+		return false, fmt.Errorf("unable to load idp cert file : %s", err)
 	}
 
 	if iss == authTokenIssuerEnvVar {
-		certificateInUse, err = readCert(dockerAuthCertEnvVar, execId)
+		certificateInUse, err = readCert(dockerAuthCertEnvVar, logger, execId)
 		if err != nil {
-			log.Printf("[%s] Unable to load docker auth file : %s\n", execId, err)
+			return false, fmt.Errorf("unable to load docker auth file : %s", err)
 		}
 	}
 
-	tokenValidity, err := validateToken(token, certificateInUse, execId)
+	tokenValidity, err := validateToken(token, certificateInUse, logger, execId)
 	if err != nil {
-		log.Printf("[%s] Token is not valid : %s\n", execId, err)
-		return false
+		return false, fmt.Errorf("token is not valid : %s", err)
 	}
-	log.Printf("[%s] Signature verified\n", execId)
+	logger.Debugf("[%s] Signature verified", execId)
 
 	if tokenValidity {
-		log.Printf("[%s] User successfully authenticated\n", execId)
-		return true
+		logger.Debugf("[%s] User successfully authenticated", execId)
+		return true, nil
 	} else {
-		log.Printf("[%s] Authentication failed\n", execId)
-		return false
+		logger.Debugf("[%s] Authentication failed", execId)
+		return false, nil
 	}
 }
 
 // isJWT checks whether the token is jwt token or access token.
-func isJWT(execId string) bool {
+func isJWT() (bool, error) {
```

Comment:
This function doesn't have a meaningful name. This is supposed to check whether we need to validate JWT as well. We can remove this as well since this is no longer used.

Reply:
Fixed by #159